Python3 port

rhaamo opened the pull request:
Things done (non exhaustive list):
- move python requirements to `requirements.txt` (including aprs-python)
- update docs for py3
- fix some `print()`
- references to python2 changed for python
- singleton class calls fixes
- removed some unused imports

The map works, history seems too, I used the aprs-is network in a collector to get some datas for debugging and so far after one hour things seems ok for me.

I have waiting for next PRs: working systemd unit files, nginx & php-fpm config & dockerfile

qvarforth commented:
Thank you for your contribution!

I will look at the changes as soon as possible. I will probably set up a server to test them too. 

It may take some time before I have the opportunity to merge the changes as I have a lot to do in the near future. I apologize for that.

rhaamo commented:
Great thanks! No problem for the time to merge.

I've been running it on Python 3.8 and Nginx/PHP-FPM since 8 days now with one collector from my aprsc, and one collector from aprs-is global and so far no issue.

That make me thought that since I have renamed `trackdirect-python2.dockerfile` to `trackdirect-python.dockerfile` we will need peterus to update its docker build part for release images.

After this PR I will make somes with what I have worked with too:
- systemd services for collectors, wsserver, remover and ogn updater (both with timers)
- Nginx+PHP-FPM (90% working, I just have one non-working rewrite-rule for now)
- maybe some docs updates (virtualenv, rights, split in dedicated files, etc.)
 
Also if you are interested I can make a whole PR for [black](https://github.com/psf/black) and [flake8](https://flake8.pycqa.org/en/latest/) configs and fixes/code styling.

peterus commented on server/scripts/collector.sh:
```diff
@@ -17,6 +17,6 @@ else
 
     export PYTHONPATH=$PYTHONPATH:$CURRENTDIR/../trackdirect
     cd $CURRENTDIR/..
```
can be deleted

peterus commented on server/scripts/remover.sh:
```diff
@@ -16,6 +16,6 @@ else
 
     export PYTHONPATH=$PYTHONPATH:$CURRENTDIR/../trackdirect
     cd $CURRENTDIR/..
```
can be deleted

peterus commented on server/scripts/stationremover.sh:
```diff
@@ -17,6 +17,6 @@ else
 
     export PYTHONPATH=$PYTHONPATH:$CURRENTDIR/../trackdirect
     cd $CURRENTDIR/..
```
can be deleted

peterus commented on server/scripts/wsserver.sh:
```diff
@@ -15,6 +15,6 @@ else
 
     export PYTHONPATH=$PYTHONPATH:$CURRENTDIR/../trackdirect
     cd $CURRENTDIR/..
```
can be deleted

peterus commented on trackdirect-python.dockerfile:
```diff
@@ -0,0 +1,15 @@
+  python3 \
+  python3-dev \
+  python3-pip \
+  git \
+  curl \
+  wget \
```
not used anymore. not sure about wget, will test and report.

peterus replied:
after testing, this is the correct apt-get call:

```
RUN apt-get update && apt-get install -y \
  python3 \
  python3-dev \
  python3-pip \
  python-is-python3 \
  git \
  && rm -rf /var/lib/apt/lists/*
```

peterus commented:
added some comments, will test the docker parts.

peterus commented on docker-compose-rel.yml:
```diff
@@ -34,7 +34,7 @@ services:
       - "aprsc"
 
   heatmaps:
-    image: peterus/trackdirect-python2:latest
+    image: peterus/trackdirect-python:latest
     restart: always
     volumes:
       - $PWD/config/trackdirect.ini:/root/trackdirect/config/trackdirect.ini
```
please remove `heatmap` service as it is not used anymore

peterus commented:
just finished my review. except the last changes I suggested, everything is working fine! thanks for this big changes!
@qvarforth go from my side ;)

qvarforth commented:
Thanks again for you for your contribution! I'm merging this now.

@peterus I can fix your suggested changes after it has been merged.
